Classes/Camera/CameraController.swift

```diff
@@ -297,27 +312,47 @@ public class CameraController: UIViewController, MediaClipsEditorDelegate, Camer
     
     override public func viewDidAppear(_ animated: Bool) {
         super.viewDidAppear(animated)
+        guard cameraInputController.willCloseSoon == false else {
+            return
+        }
+        if segments.isEmpty == false && showPreview {
+            showPreviewWithSegments(segments, selected: segments.startIndex, animated: false)
+            showPreview = false
+        }
         if delegate?.cameraShouldShowWelcomeTooltip() == true && cameraPermissionsViewController.hasFullAccess() {
             showWelcomeTooltip()
         }
     }
 
     // MARK: - navigation
+    private var segments: [CameraSegment] = []
+    private var showPreview: Bool = false
     
-    private func showPreviewWithSegments(_ segments: [CameraSegment]) {
+    private func showPreviewWithSegments(_ segments: [CameraSegment], selected: Array<CameraSegment>.Index, edits: [Data?]? = nil, animated: Bool = true) {
+        guard view.superview != nil else {
+            return
+        }
         modeAndShootController.dismissTooltip()
-        let controller = createNextStepViewController(segments)
-        self.present(controller, animated: true)
+        cameraInputController.stopSession()
+        let controller = createNextStepViewController(segments, selected: selected, edits: edits)
+        self.present(controller, animated: animated)
         mediaPlayerController = controller
         if controller is EditorViewController {
             analyticsProvider?.logEditorOpen()
         }
     }
     
-    private func createNextStepViewController(_ segments: [CameraSegment]) -> MediaPlayerController {
+    private func createNextStepViewController(_ segments: [CameraSegment], selected: Array<CameraSegment>.Index, edits: [Data?]?) -> MediaPlayerController {
         let controller: MediaPlayerController
-        if settings.features.editor {
-            controller = createEditorViewController(segments)
+        if settings.features.multipleExports {
```

Comment:
[nitpick] maybe we should check `settings.features.editor` here as well? I'm not sure how we want to handle nested flags going forward though.